File: source/downloads/code/LeetCode/Java/Find-the-Difference.java
public class Solution {
    public char findTheDifference(String s, String t) {
        int n = t.length();
        char ret = t.charAt(n-1);
        for(int i = 0; i < n-1;i++){
            ret ^= t.charAt(i);
            ret ^= s.charAt(i);
        }
        return ret;
    }
}

File: downloads/code/LeetCode/Python/find-the-difference.py
class Solution(object):
    def findTheDifference(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: str
        """
        if s == None:
            return t

        s = sorted(s)
        t = sorted(t)

        for i in range(len(s)):
            if t[i] != s[i]:
                return t[i]
        return t[-1]        
